lib/src/model/settings/board_preferences.dart

```diff
@@ -347,6 +354,19 @@ enum ClockPosition {
   };
 }
 
+enum CastlingMethod {
+  kingOverRook,
+  kingTwoSquares,
+  either;
+
+  String castlingMethodl10n(BuildContext context, CastlingMethod castlingMethod) =>
```

Comment:
The preferred way is:
```dart
String l10n(AppLocalizations l10n) => switch (this) {
         CastlingMethod.kingOverRook => l10n.preferencesCastleByMovingOntoTheRook,
        CastlingMethod.kingTwoSquares => l10n.preferencesCastleByMovingTwoSquares,
        CastlingMethod.either => 'Either', //TODO l10n string
}
```

We don't want to pass `BuildContext` and other enums in the app have the same method named `l10n`